plugins/modules/system/xfconf.py

```diff
@@ -112,195 +111,157 @@
+    module = dict(
+        argument_spec=dict(
+            state=dict(default=SET,
+                       choices=VALID_STATES,
+                       type='str'),
+            channel=dict(required=True, type='str'),
+            property=dict(required=True, type='str'),
+            value_type=dict(required=False, type='list',
+                            elements='str', choices=VALID_VALUE_TYPES),
+            value=dict(required=False, type='list', elements='raw'),
+            force_array=dict(default=False, type='bool', aliases=['array']),
+        ),
+        required_if=[('state', SET, ['value', 'value_type'])],
+        required_together=[('value', 'value_type')],
+        supports_check_mode=True,
+    )
+
+    default_state = SET
+
+    command = 'xfconf-query'
+    command_args_formats = dict(
+        channel=dict(fmt=('--channel', '{0}'),),
+        property=dict(fmt=('--property', '{0}'),),
+        is_array=dict(fmt="--force-array", style=ArgFormat.BOOLEAN),
+        reset=dict(fmt="--reset", style=ArgFormat.BOOLEAN),
+        create=dict(fmt="--create", style=ArgFormat.BOOLEAN),
+        values_and_types=dict(fmt=values_fmt)
+    )
+
+    def update_xfconf_output(self, **kwargs):
+        self.update_output(**kwargs)
+        self.update_facts(**kwargs)
 
-    def sanitize(self):
-        self.previous_value = self.get()
+    def __init_module__(self):
+        self.does_not = 'Property "{0}" does not exist on channel "{1}".'.format(self.module.params['property'],
+                                                                                 self.module.params['channel'])
+        self.vars.previous_value = self._get()
 
-        if self.value is None and self.value_type is None:
-            return
-        if (self.value is None) ^ (self.value_type is None):
-            raise XFConfException('Must set both "value" and "value_type"')
+    def process_command_output(self, rc, out, err):
+        if err.rstrip() == self.does_not:
+            return None
+        if rc or len(err):
+            raise XFConfException('xfconf-query failed with error (rc={0}): {1}'.format(rc, err))
 
+        result = out.rstrip()
+        if "Value is an array with" in result:
+            result = result.split("\n")
+            result.pop(0)
+            result.pop(0)
+
+        return result
+
+    def _get(self):
+        return self.run_command(params=('channel', 'property'))
+
+    def state_get(self):
+        self.vars.value = self.vars.previous_value
+
+    def state_absent(self):
+        self.vars.value = None
+        self.run_command(params=('channel', 'property', 'reset'))
+
+    def state_present(self):
         # stringify all values - in the CLI they will all be happy strings anyway
         # and by doing this here the rest of the code can be agnostic to it
-        self.value = [str(v) for v in self.value]
+        self.vars.value = list([str(v) for v in self.module.params['value']])
```

Comment:
```suggestion
        self.vars.value = [str(v) for v in self.module.params['value']]
```
It's already a list.